pkgs/development/tools/rust/cargo-embed/default.nix

```diff
@@ -16,7 +23,7 @@ rustPlatform.buildRustPackage rec {
   cargoSha256 = "00p2rwqrax99kwadc8bfq8pbcijals2nzpx43wb03kwxl4955wn9";
 
   nativeBuildInputs = [ pkg-config rustfmt ];
-  buildInputs = [ libusb1 libftdi1 ];
+  buildInputs = [ libusb1 libftdi1 ] ++ lib.optionals stdenv.isDarwin [ AppKit ];
```

Comment:
```suggestion
  buildInputs = [ libusb1 libftdi1 ] ++ lib.optional stdenv.isDarwin AppKit;
```

Reply:
I do like the optionals better of the optional for the simple reason that I find it more extensible. If I anything needs to be added, then it's a smaller change.
If you don't feel strongly about it, I would like to keep it.

Reply:
Doesn't matter really what is used. Either is fine.